std/net/curl.d

```diff
@@ -2879,7 +2879,7 @@ struct HTTP
     /** The headers read from a successful response.
      *
      */
-    @property string[string] responseHeaders()
+    @property string[string] responseHeaders() return
     {
         return p.headersIn;
```

Comment:
As far as the compiler is concerned, you could have used `malloc` to create the associative array (even though currently that's far from straightforward). The only thing that it cares about is that you're returning a reference to one of its members. `scope` and `return` are orthogonal to `@nogc` AFAIU.

The strange thing is that it looks like the inferred `scope` of `this` is transitive, because you're not returning a reference to a field, but a reference to the object pointed by the field. @WalterBright can you shed some light on this?